org.testeditor.web.dropwizard.xtext/src/main/java/org/testeditor/web/xtext/index/changes/GitBasedChangeDetector.xtend

```diff
@@ -21,7 +21,17 @@ class GitBasedChangeDetector implements ChangeDetector {
 
 	var String lastUpdatedAtRevision = null
 
-	override detectChanges(ResourceSet resourceSet, String[] paths, ChangedResources accumulatedChanges) {
+	override ChangedResources collectFull(ResourceSet resourceSet, String[] paths, ChangedResources accumulatedChanges) {
```

Comment:
I think `detectChanges` does the same _if_ `lastUpdatedAtRevision === null`. So, you should be able to just do
```
lastUpdatedAtRevision = null
return detectChanges(…)
```
(or introduce a `reset` method, so `detectChanges` can be used directly?)